presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeWindowQueries.java

```diff
@@ -50,23 +55,58 @@ protected void createTables()
             "PARTITION BY custkey, orderkey, shippriority",
             "PARTITION BY orderkey, custkey");
 
-    protected List<String> getRankingQueries(String rankingFunction, boolean orderBy)
+    private static final List<String> FRAME_CLAUSES = Arrays.asList(
+            // Frame clauses in RANGE mode.
+            "RANGE UNBOUNDED PRECEDING",
+            "RANGE CURRENT ROW",
+            "RANGE BETWEEN UNBOUNDED PRECEDING and UNBOUNDED FOLLOWING",
+
+            // Frame clauses in ROWS mode.
+            "ROWS UNBOUNDED PRECEDING",
+            "ROWS BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN 5 PRECEDING AND CURRENT ROW",
+            "ROWS BETWEEN 5 PRECEDING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN CURRENT ROW AND 5 FOLLOWING",
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND 5 FOLLOWING",
+            "ROWS BETWEEN orderkey PRECEDING AND CURRENT ROW",
+            "ROWS BETWEEN orderkey PRECEDING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN CURRENT ROW AND orderkey FOLLOWING",
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND orderkey FOLLOWING",
+
+            // Frame clauses with empty frames.
+            "ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING",
+            "ROWS BETWEEN 1 PRECEDING AND 4 PRECEDING",
+            "ROWS BETWEEN 1 FOLLOWING AND UNBOUNDED FOLLOWING",
+            "ROWS BETWEEN 4 FOLLOWING AND 1 FOLLOWING");
+
+    protected List<String> getQueries(String function, FunctionType functionType)
     {
         ImmutableList.Builder<String> queries = ImmutableList.builder();
         List<String> columnProjections = Arrays.asList("orderkey, orderdate, totalprice");
-        List<String> overClauses = orderBy ? OVER_CLAUSES_WITH_ORDER_BY : OVER_CLAUSES_WITHOUT_ORDER_BY;
+        List<String> overClauses = new ArrayList<>(OVER_CLAUSES_WITH_ORDER_BY);
+        List<String> frameClauses = FRAME_CLAUSES;
+        if (functionType == FunctionType.VALUE) {
+            overClauses.addAll(OVER_CLAUSES_WITHOUT_ORDER_BY);
+        } 
 
         for (String columnProjection : columnProjections) {
             for (String overClause : overClauses) {
-                queries.add(String.format("SELECT %s, %s OVER (%s) AS rnk FROM orders", columnProjection, rankingFunction, overClause));
+                if (functionType == FunctionType.VALUE) {
+                    for (String frameClause : frameClauses) {
+                        queries.add(String.format("SELECT %s, %s OVER (%s %s) AS val FROM orders", columnProjection, function, overClause, frameClause));
```

Comment:
I feel this would affect the readability of the query, but it would send fewer queries to be executed which is better. Grouped 5 function calls into a single Sql query as suggested.